common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1233,7 +1233,12 @@ def submit_problem(self, data, override_time=False):
             # without a stack trace
             else:
                 # Translators: {msg} will be replaced with a problem's error message.
-                msg = inst.message
+                try:
+                    # only return the error value of the exception
+                    message = inst.message.split("\\n")[-2].split(": ", 1)[1]
+                    msg = cgi.escape(message)
+                except IndexError:
+                    msg = inst.message
```

Comment:
When I test this on my devstack, inst.message is an empty string, and still results in the student seeing an empty message.  I'm not sure what's different in your environment.  inst.args[0] seems to have the full message though.

I see discussion below about inst.args[0].  Did it used to use that?